test/E2E/E2ETest.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Xunit;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.ProjectModel;
+
+namespace ConsoleApplication
+{
+    public class E2ETest
+    {
+        private static readonly string EXPECTED_OUTPUT = "Hello World!";
+        private static readonly string TEST_PROJECT_NAME = "hellotest";
+        private static readonly string OUTPUT_FOLDER_NAME = "testbin";
+
+        public static void Main()
+        {
+            Console.WriteLine("Dummy Entrypoint.");
+        }
+
+        [Fact]
+        public static void TestE2E()
+        {
```

Comment:
I think this test better be refactored into multiple tests for each command. Even if one command fails the remaining commands are not run and their status is unknown. 

I suggest putting dotnet-restore inside the constructor so that it invoked for each test. And each test has only one command to test.